### PR TITLE
target app configurable, default focus

### PR DESCRIPTION
**Pull request by enola-dkfz**

### CODE REVIEW

**Comment by enola-dkfz** [comment]
fixes https://github.com/samply/spot/issues/36

File: src/config.rs [hunk 1]
```diff
@@ -37,12 +37,16 @@ pub struct Config {
...
+    #[clap(long, env, default_value = "http://localhost:8066")]
     pub prism_url: Url,
 
     /// Path to a file which will contain the query logs
     #[clap(long, env, value_hint = clap::ValueHint::FilePath)]
     pub log_file: Option<PathBuf>,
+
+    /// Target_application_name
+    #[clap(long, env, value_parser, default_value = "focus")]
+    pub target: String,
```
Comment (TKussel) [inline src/config.rs:49]:
```suggestion
    pub target_app: String,
```

File: src/beam.rs [hunk 2]
```diff
@@ -6,9 +6,10 @@ pub fn create_beam_task(
     target_sites: Vec<String>,
     query: String,
 ) -> TaskRequest<RawString> {
+    let target = &CONFIG.target;
```
Comment (TKussel) [inline src/beam.rs:9]:
```suggestion
    let target_app = &CONFIG.target_app;
```

File: src/beam.rs [hunk 3]
```diff
@@ -6,9 +6,10 @@ pub fn create_beam_task(
...
     let proxy_id = CONFIG.beam_app_id.proxy_id();
     let broker_id = proxy_id.as_ref().split_once('.').expect("Invalid beam id in config").1;
-    let to = target_sites.into_iter().map(|site| AppId::new_unchecked(format!("focus.{site}.{broker_id}"))).collect();
+    let to = target_sites.into_iter().map(|site| AppId::new_unchecked(format!("{target}.{site}.{broker_id}"))).collect();
```
Comment (TKussel) [inline src/beam.rs:12]:
```suggestion
    let to = target_sites.into_iter().map(|site| AppId::new_unchecked(format!("{target_app}.{site}.{broker_id}"))).collect();
```

File: README.md [hunk 4]
```diff
@@ -32,6 +32,8 @@ Optional environment variables:
     URL to prism, if catalogue-url is not stated, this is never used [env: PRISM_URL=] [default: http://localhost:8066]
 --bind-addr <BIND_ADDR>
     The socket address this server will bind to [env: BIND_ADDR=] [default: 0.0.0.0:8055]
+--target <TARGET>
```
Comment (TKussel) [inline README.md:35]:
```suggestion
--target-app <TARGET_APP>
```

File: README.md [hunk 5]
```diff
@@ -32,6 +32,8 @@ Optional environment variables:
...
+    Target_application_name [env: TARGET=] [default: focus]
```
Comment (TKussel) [inline README.md:36]:
```suggestion
    Target_application_name [env: TARGET_APP=] [default: focus]
```

File: README.md [hunk 6]
```diff
@@ -7,7 +7,7 @@ Spot can be run locally with the provided [docker-compose](./docker-compose.yml)
 
 Spot can also be run from command line stating command line parameters, like so:
 ```bash
-cargo run -- --beam-proxy-url http://localhost:8081 --beam-app-id app1.proxy1.broker --beam-secret App1Secret --cors-origin any --bind-addr 127.0.0.1:8055 --catalogue-url https://raw.githubusercontent.com/samply/lens/main/packages/demo/public/catalogues/catalogue-dktk.json
+cargo run -- --beam-proxy-url http://localhost:8081 --beam-app-id app1.proxy1.broker --beam-secret App1Secret --cors-origin any --bind-addr 127.0.0.1:8055 --catalogue-url https://raw.githubusercontent.com/samply/lens/main/packages/demo/public/catalogues/catalogue-dktk.json --target app1
```
Comment (TKussel) [inline README.md:10]:
```suggestion
cargo run -- --beam-proxy-url http://localhost:8081 --beam-app-id app1.proxy1.broker --beam-secret App1Secret --cors-origin any --bind-addr 127.0.0.1:8055 --catalogue-url https://raw.githubusercontent.com/samply/lens/main/packages/demo/public/catalogues/catalogue-dktk.json --target-app app1
```

**Review by TKussel — APPROVED** [review]
THANKS! However, I think `target` is too unspecific. See my suggestion.

**Comment by enola-dkfz** [comment]
Great rename Tobias, would you be so kind to do them it in Samply/Prism as well?

**Pull request merged**